mempool/mempool.go

```diff
@@ -497,5 +497,6 @@ func (cache *txCache) Push(tx types.Tx) bool {
 func (cache *txCache) Remove(tx types.Tx) {
 	cache.mtx.Lock()
 	delete(cache.map_, string(tx))
+	// so the list grows unboundedly? shouldn't we remove tx from the list here too?
```

Comment:
the list is capped in `Push`, where we remove from the map and list. So it's ok if we don't remove from the list in Remove, but I don't really see why we wouldn't (maybe performance ?).

Reply:
![1](https://user-images.githubusercontent.com/1282182/40194610-173d54c4-5a1c-11e8-97f4-9866090a2a7d.png)

assuming an attacker creates txs that do not pass CheckTx, list grows unboundedly.

Reply:
I don't understand why another transaction got added after CheckTx fails.

Reply:
I was wrong.